tests/providers/amazon/aws/sensors/test_emr_job_flow.py

```diff
@@ -276,3 +277,20 @@ def test_different_target_states(self):
             # make sure it was called with the job_flow_id
             calls = [mock.call(ClusterId="j-8989898989")]
             self.mock_emr_client.describe_cluster.assert_has_calls(calls)
+
+    @mock.patch("airflow.providers.amazon.aws.sensors.emr.EmrJobFlowSensor.poke")
+    def test_sensor_defer(self, mock_poke):
+        sensor = EmrJobFlowSensor(
+            task_id="test_task",
+            poke_interval=0,
+            job_flow_id="j-8989898989",
+            aws_conn_id="aws_default",
+            target_states=["RUNNING", "WAITING"],
+            deferrable=True,
+        )
+        mock_poke.return_value = False
+        with pytest.raises(TaskDeferred) as exc:
+            sensor.execute(context=None)
+        assert isinstance(
+            exc.value.trigger, EmrTerminateJobFlowTrigger
+        ), "Trigger is not a EmrTerminateJobFlowTrigger "
```

Comment:
Perhaps we can enhance the message by including the trigger in the print.
```suggestion
        ), f"{exc.value.trigger} is not a EmrTerminateJobFlowTrigger "
```